src/replit_river/session.py

```diff
@@ -96,7 +124,11 @@ def increment_and_get_heartbeat_misses() -> int:
                 self.session_id,
                 self._transport_options.heartbeat_ms,
                 self._transport_options.heartbeats_until_dead,
-                lambda: self._state,
+                lambda: (
+                    self._state
+                    if self._ws_wrapper.ws_state == WsState.OPEN
+                    else SessionState.CONNECTING
```

Comment:
a session wont always have a ws right? what about the SessionState.PENDING state?

Reply:
this is v1 as well. To Faris' point, it's hard to remember that the only v2 changes that matter are in the `v2` package